ruby/lib/libdatadog.rb

```diff
@@ -64,4 +64,12 @@ def self.path_to_crashtracking_receiver_binary
 
     File.absolute_path("#{pkgconfig_folder}/../../bin/libdatadog-crashtracking-receiver")
   end
+
+  def self.ld_library_path
+    pkgconfig_folder = self.pkgconfig_folder
+
+    return unless pkgconfig_folder
+
+    File.absolute_path("#{pkgconfig_folder}/../")
```

Comment:
I trust you know what you are doing... 😉

Reply:
Ahaha thanks! To be clear, what this does is take the structure of our binary builds:

```
% tree
.
├── bin
│   └── libdatadog-crashtracking-receiver
├── cmake
│   └── DatadogConfig.cmake
├── include
│   └── datadog
│       ├── blazesym.h
│       ├── common.h
│       ├── profiling.h
│       └── telemetry.h
├── lib
│   ├── libdatadog_profiling.a
│   ├── libdatadog_profiling.so
│   ├── libdatadog_profiling.so.debug
│   └── pkgconfig
│       ├── datadog_profiling.pc
│       ├── datadog_profiling-static.pc
│       └── datadog_profiling_with_rpath.pc
├── LICENSE
├── LICENSE-3rdparty.yml
└── NOTICE
```

and encodes the knowledge that the .so file is in the parent folder to the pkgconfig file.

I could've implemented this another way (e.g. search for the so file directly or something), but since this is internal to the libdatadog helper, and we already have another helper that does this relative find thing, I just went with it.

(Since on the Ruby library side we only call these methods and don't encode any assumptions on the folder structure, we can always update the helpers and move stuff around whenever we want to).